nodes/test/cobalt/test_dataset_cobalt.py

```
@@ -0,0 +1,844 @@
+            sys.modules.pop(name, None)
+        else:
+            sys.modules[name] = orig
+
+
+# Snapshot before mocks are installed, then install mocks so node imports succeed.
+_snapshot_modules()
+_install_mocks()
+
+
+@pytest.fixture(autouse=True, scope='session')
+def _restore_modules():
```

Ruff (0.15.10)

[warning] Missing return type annotation for private function `_restore_modules`

(ANN202)